travis.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -uexo pipefail
+
+declare -A compilers=(
+ ["dmd-2.069.2"]="DMD64 D Compiler v2.069.2"
+ ["dmd-2.071.2"]="DMD64 D Compiler v2.071.2"
+ ["dmd-2016-10-19"]="DMD64 D Compiler v2.073.0-master-878b882"
+ ["dmd-master-2016-10-24"]="DMD64 D Compiler v2.073.0-master-ab9d712"
+ ["ldc-1.0.0"]="LDC - the LLVM D compiler (1.0.0):"
+ ["gdc-4.9.3"]="gdc (crosstool-NG crosstool-ng-1.20.0-232-gc746732 - 20150825-2.066.1-58ec4c13ec) 4.9.3"
```

Comment:
Should test name resolutions as well. The values could be globs matched via `~=` or regexes for grep/egrep.